### PR TITLE
Dealing with two transboundary bug

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #
An out-of-bounds error in calculating the step when processing multiple cn and an out-of-bounds error in calculating the step when processing parallel reads.
## What this PR does / why we need it:

### CODE REVIEW

File: pkg/sql/compile/compile.go [hunk 1]
```diff
@@ -1266,7 +1266,7 @@ func (c *Compile) generateNodes(n *plan.Node) (engine.Nodes, error) {
 	if len(ranges) == 0 {
 		return nodes, nil
 	}
-	step := len(ranges) / len(c.cnList)
+	step := (len(ranges) / len(c.cnList)) + 1
 	if step <= 0 {
```
Comment (fengttt) [inline pkg/sql/compile/compile.go:1270]: Now step cannot be <= 0.    What is a step?

Comment (nnsgmsone) [reply pkg/sql/compile/compile.go:1270]: ranges represents how many blocks there are in total, cnList represents how many cn nodes exist in total, and then step represents how many blocks are processed by each cn node.

Comment (nnsgmsone) [reply pkg/sql/compile/compile.go:1270]: yes, now step cannot be <=0

**Review by yingfeng — APPROVED** [review]

File: pkg/sql/compile/compile.go (hunk 1 above)
Comment (fengttt) [reply pkg/sql/compile/compile.go]: This code is wrong.

if ranges is 20, len(cnList) is also 20, you break the steps to 2, 2, 2, ..., 0, 0, 0 ...?

Should this be (len(ranges) + len(c.cnList) - 1) / len(c.cnList)?

Comment (fengttt) [reply pkg/sql/compile/compile.go]: I assume the following 0, 0, ... will not crash our code.

Comment (nnsgmsone) [reply pkg/sql/compile/compile.go]: yes

**Pull request merged**